changelog.d/10326.bugfix

```diff
@@ -0,0 +1 @@
+Fix a long-standing bug where Synapse would return errors after 2<sup>31</sup> events were handled by the server.
```

Comment:
this changelog doesn't seem to match? or is this PR being deliberately merged into the overall 32-bit rollover problem?

Reply:
The latter